api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -426,6 +426,7 @@ type HostedClusterSpec struct {
 	//
 	// If a given service is not present in this list, it will be exposed publicly
 	// by default.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Services is immutable"
```

Comment:
I know its far cry, but if we can add some reasoning on why this is a good thing in the message, it would be great. 

"Serivices are immutable to/because ...."